js_modules/dagster-ui/packages/ui-core/src/assets/Stale.tsx

```diff
@@ -127,50 +175,160 @@ function groupedCauses(
+      return (
+        <>
+          <Caption>{reasonUpToDep}</Caption>
+          <Link to={assetDetailsPathForKey(dependency)}>
+            <Tag icon="asset">{dependencyName}</Tag>
+          </Link>
+        </>
+      );
+    }
 
-  const dependencyName = displayNameForAssetKey(dependency);
-  const dependencyPythonName = dependencyName.replace(/ /g, '');
-  if (reason.endsWith(`${dependencyPythonName}`)) {
-    const reasonUpToDep = reason.slice(0, -dependencyPythonName.length);
     return (
-      <Caption>
-        {` ${reasonUpToDep}`}
-        <Link to={assetDetailsPathForKey(dependency)}>{dependencyName}</Link>
-      </Caption>
+      <>
+        <Link to={assetDetailsPathForKey(dependency)}>
+          <Tag icon="asset">{dependencyName}</Tag>
+        </Link>
+        <Caption>{` ${reason} `}</Caption>
+      </>
     );
-  }
+  }, [dependency, reason]);
 
   return (
-    <Caption>
-      {` ${reason} `}(<Link to={assetDetailsPathForKey(dependency)}>{dependencyName}</Link>)
-    </Caption>
+    <Box
+      padding={{vertical: 8, horizontal: 12}}
+      flex={{direction: 'row', alignItems: 'center', gap: 4}}
+    >
+      {content}
+    </Box>
+  );
+};
+
+export const NewOrChangedInBranchTag = ({
+  changedReasons,
+  assetKey,
+}: {
+  changedReasons: ChangeReason[];
+  assetKey: AssetKeyInput;
+}) => {
+  const changes = changedReasons.filter((reason) => reason !== ChangeReason.NEW);
+
+  function getDescription(change: ChangeReason) {
+    switch (change) {
+      case ChangeReason.NEW:
+        return '';
+      case ChangeReason.CODE_VERSION:
+        return 'has a modified code version';
+      case ChangeReason.INPUTS:
+        return 'has modified dependencies';
+      case ChangeReason.PARTITIONS_DEFINITION:
+        return 'has a modified partition definition';
+    }
+  }
+  return (
+    <Popover
+      position="top"
+      isOpen={changes.length ? undefined : false}
+      content={
+        <Box flex={{direction: 'column'}}>
+          <Box padding={{horizontal: 12, vertical: 8}} border="bottom">
+            <Subtitle2>
+              {numberFormatter.format(changes.length)}{' '}
+              {ifPlural(changes.length, 'change', 'changes')} in this branch
+            </Subtitle2>
+          </Box>
+          {changes.map((change) => {
+            return (
+              <Box
+                key={change}
+                padding={{vertical: 8, horizontal: 12}}
+                flex={{direction: 'row', alignItems: 'center', gap: 4}}
+              >
+                <Tag icon="asset">{displayNameForAssetKey(assetKey)}</Tag>
+                {getDescription(change)}
+              </Box>
+            );
+          })}
+        </Box>
+      }
+      interactionKind="hover"
+      className="chunk-popover-target"
+    >
+      <BaseTag
+        fillColor={Colors.backgroundCyan()}
+        textColor={Colors.textCyan()}
+        label={changes.length ? 'Modified in branch' : 'Modified in branch'}
```

Comment:
should one of these be `New in branch`? noticed that the new asset is now being marked "Modified in branch" instead of "New in branch"